src/Admin/OptionsMenu.php

```diff
@@ -0,0 +1,483 @@
+		add_submenu_page(
+			AMP_Options_Manager::OPTION_NAME,
+			__( 'AMP Settings', 'amp' ),
+			__( 'General', 'amp' ),
+			'manage_options',
+			AMP_Options_Manager::OPTION_NAME
+		);
+
+		add_settings_section(
+			'general',
+			false,
+			'__return_false',
+			AMP_Options_Manager::OPTION_NAME
+		);
+
+		add_settings_section(
+			'validation',
+			false,
+			'__return_false',
+			AMP_Options_Manager::OPTION_NAME
+		);
+
+		add_settings_field(
+			Option::SUPPORTED_TEMPLATES,
+			__( 'Supported Templates', 'amp' ),
+			[ $this, 'render_supported_templates' ],
+			AMP_Options_Manager::OPTION_NAME,
+			'general',
+			[
+				'class' => 'amp-template-support-field',
+			]
+		);
+
+		/**
+		 * This fires when settings fields for the AMP Options menu need to be registered.
+		 *
+		 * This action is intended for internal use only, not to be used by other plugins.
+		 *
+		 * @internal
+		 */
+		do_action( 'amp_options_menu_items' );
+
+		$submenus = [
+			new AMP_Analytics_Options_Submenu( AMP_Options_Manager::OPTION_NAME ),
+		];
+
+		// Create submenu items and calls on the Submenu Page object to render the actual contents of the page.
+		foreach ( $submenus as $submenu ) {
+			$submenu->init();
+		}
+	}
+
+	/**
+	 * Provides the settings screen handle.
+	 *
+	 * @return string
+	 */
+	public function screen_handle() {
+		return sprintf( 'toplevel_page_%s', AMP_Options_Manager::OPTION_NAME );
+	}
+
+	/**
+	 * Enqueues settings page assets.
+	 *
+	 * @since 1.6.0
+	 *
+	 * @param string $hook_suffix The current admin page.
+	 */
+	public function enqueue_assets( $hook_suffix ) {
+		if ( $this->screen_handle() !== $hook_suffix ) {
+			return;
+		}
+
+		$asset_file   = AMP__DIR__ . '/assets/js/' . self::ASSET_HANDLE . '.asset.php';
+		$asset        = require $asset_file;
+		$dependencies = $asset['dependencies'];
+		$version      = $asset['version'];
+
+		wp_enqueue_script(
+			self::ASSET_HANDLE,
+			amp_get_asset_url( 'js/' . self::ASSET_HANDLE . '.js' ),
+			$dependencies,
+			$version,
+			true
+		);
+
+		wp_enqueue_style(
+			self::ASSET_HANDLE,
+			amp_get_asset_url( 'css/amp-settings.css' ),
+			[ $this->google_fonts->get_handle(), 'wp-components' ],
+			AMP__VERSION
+		);
+
+		wp_styles()->add_data( self::ASSET_HANDLE, 'rtl', 'replace' );
+
+		$theme           = wp_get_theme();
+		$is_reader_theme = in_array( get_stylesheet(), wp_list_pluck( ( new AMP_Reader_Themes() )->get_themes(), 'slug' ), true );
+
+		$js_data = [
+			'CURRENT_THEME'                      => [
+				'name'            => $theme->get( 'Name' ),
+				'description'     => $theme->get( 'Description' ),
+				'is_reader_theme' => $is_reader_theme,
+				'screenshot'      => $theme->get_screenshot(),
+				'url'             => $theme->get( 'ThemeURI' ),
+			],
+			'OPTIONS_REST_ENDPOINT'              => rest_url( 'amp/v1/options' ),
+			'READER_THEMES_REST_ENDPOINT'        => rest_url( 'amp/v1/reader-themes' ),
+			'THEME_IS_SUPPORTED'                 => in_array(
+				get_template(),
```

Comment:
Per my comment in #4990:

> Note I'm intentionally using `get_stylesheet()` here as opposed to `get_template()` because child themes of core themes are often not AMP-compatible.

This should rather be `get_stylesheet()`:

```suggestion
				get_stylesheet(),
```

This will ensure that if the user has a child theme of a core theme that it won't be declared AMP-compatible when the site owner may very well have made it incompatible. 

As a reminder, `get_template()` returns the parent theme and `get_stylesheet()` returns the child theme. If no child theme is being used, then they return the same value.

Reply:
Updated.